### PR TITLE
refactor(bpmn-model) Improve error message of BPMN features

### LINKED ISSUE #7282: Improve error message in case of uncovered BPMN features
**Is your feature request related to a problem? Please describe.**
I did report issue #7279, where I thought I found a bug because the error message did not let me think of missing feature implementation, but rather XML-related issues. So I spend some time to validated the XML file against many validators.

```
Error: rpc error: code = InvalidArgument desc = Command 'CREATE' rejected with code 'INVALID_ARGUMENT': Expected to deploy new resources, but encountered the following errors:
'/Users/maki/Downloads/hello_world.bpmn': - Element: Activity_1yyow37
    - ERROR: Elements of this type are not supported
``` 

**Describe the solution you'd like**
When Zeebe only covers a few BPMN features
(as described https://docs.camunda.io/docs/reference/bpmn-processes/bpmn-coverage#tasks)
I would prefer to see an error message, which better guides me as a user.
Considering the fact, the error message is shown in CLI tool, simple monitor, and others,
this should be much as possible self-describing.

**Describe alternatives you've considered**
May I propose an error message with more content and guidance?

Proposal
```
Error: rpc error: code = InvalidArgument desc = Command 'CREATE' rejected with code 'INVALID_ARGUMENT': Expected to deploy new resources, but encountered the following errors:
'/Users/maki/Downloads/hello_world.bpmn': - Element: Activity_1yyow37
    - ERROR: An un-supported element found. The XML is valid, but currently, just a sub-set of BPMN elements are supported, see the documentation for more details: https://docs.camunda.io/docs/reference/bpmn-processes/bpmn-coverage
```

**Additional context**
See #7279

### PR DESCRIPTION
**Pull request by sccalabr**
## Description

Improve error message in case of uncovered BPMN features by linking error message to documentation. 

## Related issues

closes #7282

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [X ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by saig0** [comment]
@sccalabr thank you for your contribution. :tada:  I'll have a look at your PR in the next few days. :eyes:

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
Thanks for your contribution @sccalabr! 🚀 Sorry I took a little while to look at it.

I have a suggestion about rephrasing the message. Please let me know what you think of it!

File: bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/FlowElementValidator.java [hunk 1]
```diff
@@ -89,9 +89,9 @@ public void validate(
         && !NON_EXECUTABLE_ELEMENT_TYPES.contains(elementType)) {
       validationResultCollector.addError(
           0,
-          "An un-supported element found. The XML is valid, but "
-              + "currently, just a sub-set of BPMN elements are supported, see the documentation for "
-              + "more details: https://docs.camunda.io/docs/reference/bpmn-processes/bpmn-coverage");
+          "Elements of type 'elementType.getSimpleName()' are currently not supported. Please refer "
+              + "to the documentation for a list of supported elements: "
+              + "https://docs.camunda.io/docs/reference/bpmn-processes/bpmn-coverage");
```
Comment (remcowesterhoud) [inline bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/FlowElementValidator.java:94]: ❌ Maybe I should have clarified myself better 😄 We don't want to actually return `elementType.getSimpleName()` in the message. We want to be executed.

```suggestion
          String.format(
              "Elements of type '%s' are currently not supported. Please refer "
                  + "to the documentation for a list of supported elements: "
                  + "https://docs.camunda.io/docs/reference/bpmn-processes/bpmn-coverage",
              elementType.getSimpleName()));
```

**Review by remcowesterhoud — CHANGES_REQUESTED** [review]
I'm sorry I wasn't clear enough with my previous comment. I have left another one that should clear things up.

**Review by remcowesterhoud — APPROVED** [review]
Thanks @sccalabr! 🚀 

This pull request counts toward the Hacktoberfest challenge. If you contributed four pull requests then you can complete your Camunda Hacktoberfest challenge [here](https://camunda.com/hacktoberfest2022/).

bors merge

**Pull request merged**